.github/workflows/ci.yaml

```
@@ -11,30 +11,42 @@ on:
     steps:
       - uses: actions/checkout@v4
         with:
           submodules: recursive
       - uses: Swatinem/rust-cache@v2
         with:
-          prefix-key: v1-rust
+          prefix-key: v1-rust-${{ matrix.os }}
           shared-key: debug
           cache-all-crates: true
       - name: Setup Rust
         uses: actions-rs/toolchain@v1
```

actionlint (1.7.7)

the runner of "actions-rs/toolchain@v1" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)